lib/elixir/lib/calendar.ex

```diff
@@ -1420,4 +1505,33 @@ defmodule DateTime do
       calendar.to_string(dt)
     end
   end
+
+  @doc """
+  Compares two `DateTime` structs. Returns :gt if first datetime is later than the second
+  and :lt for vice versa. If the two datetimes are equal :eq is returned
+  Note that both utc and stc offsets will be taken into account when comparison is done.
+
+  ## Examples
+
+     iex> dt1 = %DateTime{year: 2000, month: 2, day: 29, zone_abbr: "AMT",
+     ...>                hour: 23, minute: 0, second: 7, microsecond: {0, 0},
+     ...>                utc_offset: -14400, std_offset: 0, time_zone: "America/Manaus"}
+     iex> dt2 = %DateTime{year: 2000, month: 2, day: 29, zone_abbr: "CET",
+     ...>                hour: 23, minute: 0, second: 7, microsecond: {0, 0},
+     ...>                utc_offset: 3600, std_offset: 0, time_zone: "Europe/Warsaw"}
+     iex> DateTime.compare(dt1, dt2)
+     :gt
+  """
+  @spec compare(DateTime.t, DateTime.t) :: :lt | :eq | :gt
+  def compare(%DateTime{} = datetime_1, %DateTime{} = datetime_2) do
+    do_compare(to_unix(datetime_1, :microsecond), to_unix(datetime_2, :microsecond))
+  end
+
+  def do_compare(datetime_usec_1, datetime_usec_2) do
```

Comment:
just another nitpicking comment. It's pretty much the (unwritten) standard in the Elixir project to name arguments without an underscore preceding the number. So it should read:

    def do_compare(datetime_usec1, datetime_usec2) do

thank you